src/map/battle.cpp

```diff
@@ -6474,12 +6491,42 @@ static void battle_calc_attack_post_defense(struct Damage* wd, struct block_list
 	struct status_data *sstatus = status_get_status_data(src);
 
 	// Post skill/vit reduction damage increases
-	if( sc ) { // SC skill damages
-		if(sc->getSCE(SC_AURABLADE)
 #ifndef RENEWAL
-				&& skill_id != LK_SPIRALPIERCE && skill_id != ML_SPIRALPIERCE
+	//Refine bonus
+	if (sd && battle_skill_stacks_masteries_vvs(skill_id) && skill_id != MO_INVESTIGATE && skill_id != MO_EXTREMITYFIST) { // Counts refine bonus multiple times
+		ATK_ADD2(wd->damage, wd->damage2, sstatus->rhw.atk2, sstatus->lhw.atk2);
+		ATK_ADD2(wd->basedamage, wd->basedamage2, sstatus->rhw.atk2, sstatus->lhw.atk2);
+	}
+
+	//After DEF reduction, damage can be negative, refine bonus works against that value
+	//After refinement bonus was applied, damage is capped to 1, then masteries are applied
+	if (is_attack_right_handed(src, skill_id) && wd->damage < 1) wd->damage = 1;
+	if (is_attack_left_handed(src, skill_id) && wd->damage2 < 1) wd->damage2 = 1;
+	if (is_attack_right_handed(src, skill_id) && wd->basedamage < 1) wd->basedamage = 1;
+	if (is_attack_left_handed(src, skill_id) && wd->basedamage2 < 1) wd->basedamage2 = 1;
```

Comment:
```suggestion
	if (is_attack_right_handed(src, skill_id)) {
		if (wd->damage < 1)
			wd->damage = 1;
		if (wd->basedamage < 1)
			wd->basedamage = 1;
	}
	if (is_attack_left_handed(src, skill_id)) {
		if (wd->damage2 < 1)
			wd->damage2 = 1;
		if (wd->basedamage2 < 1)
			wd->basedamage2 = 1;
	}
```
Is it easier this way? We also only make singular calls to `is_attack_<left/right>_handed()`.

Reply:
Stretches out the code a bit, maybe we should bundle this into a battle_min_damage function.